clients/TypeScript/packages/client/src/ChainSync/ChainSyncClient.ts

```diff
@@ -79,17 +63,27 @@ export const createChainSyncClient = async (options?: {
           },
           requestNext (options) {
             ensureSocketIsOpen(socket)
-            socket.send(JSON.stringify({
-              ...baseRequest,
-              methodname: 'RequestNext',
-              mirror: options?.mirror
-            } as Ogmios['RequestNext']))
+            return requestNext(socket, options)
           },
           shutdown: () => new Promise(resolve => {
             ensureSocketIsOpen(socket)
             socket.once('close', resolve)
             socket.close()
-          })
+          }),
+          startSync: async (points, requestBuffer) => {
```

Comment:
Question: I am not sure how WebSockets handles event handlers subscription behind the scene but, should this function throw if called without any handlers defined for on 'rollForward' and on 'rollBackward' ?

Reply:
I'll move the handlers to a parameter of the upper function, as there's no longer a need to register the handlers after client construction.

Reply:
Done in 15ef6a79bef920e02e7fed4ff4e0efcb4535eacc